src/main/webapp/script.js

```diff
@@ -30,7 +30,20 @@ async function getAuthToken() {
 }
 
 function getIdFromURL(url) {
-  var start = url.lastIndexOf('/') + 1;
-  var end = url.indexOf('?', start);
-  return url.slice(start, end);
+  if (url.search("open.spotify.com\/playlist\/") != -1) {
+    var start = url.lastIndexOf('/') + 1;
+    var end = url.length;
+    if (url.search('\\?') != -1) {
+     console.log('? detected');
+     end = url.indexOf('?', start);  
+    }
+    return url.slice(start, end);
+  } else {
+    console.log(url + "is not a valid Spotify URL");
```

Comment:
since we are switching to youtube now, do we want to update this code to get the playlist ID from a youtube playlist link? Or is that planned in a follow up PR?

Reply:
IMO I think it's okay to to leave this as is and update it later when we switch to Youtube, that way we don't drag out getting this PR merged too long.